lib/streamlit/elements/widgets/button.py

```diff
@@ -512,6 +539,75 @@ def _link_button(
 
         return self.dg._enqueue("link_button", link_button_proto)
 
+    def _page_link(
+        self,
+        page: str,
+        *,  # keyword-only arguments:
+        label: Optional[str] = None,
+        icon: Optional[str] = None,
+        help: Optional[str] = None,
+        disabled: bool = False,
+        use_container_width: Optional[bool] = None,
+    ) -> "DeltaGenerator":
+        page_link_proto = PageLinkProto()
+        page_link_proto.disabled = disabled
+
+        if label is not None:
+            page_link_proto.label = label
+
+        if icon is not None:
+            page_link_proto.icon = validate_emoji(icon)
+
+        if help is not None:
+            page_link_proto.help = dedent(help)
+
+        if use_container_width is not None:
+            page_link_proto.use_container_width = str(use_container_width)
+
+        # Handle external links:
+        if page.startswith("http://") or page.startswith("https://"):
+            if label is None or label == "":
+                raise StreamlitAPIException(
+                    f"The label param is required for external links used with st.page_link - please provide a label."
+                )
+            else:
+                page_link_proto.page = page
```

Comment:
it looks like the `page` argument is only used for `http://`/`https://` links, so it feels a little bit weird that it is a mandatory argument (see also comment about naming above). If only such links can be handled, should it be a check in the very beginning of the method?

Reply:
`page` argument used for both MPA pages and external links (thus mandatory) - see for loop ~line 600